src/componentsNext/Pagination/describePagination.ts

```diff
@@ -0,0 +1,143 @@
+type PaginationShape = (number | "...")[];
+
+/**
+ *
+ * describePagination is a helper function to describe a pagination
+ * depending on the current page and the amount of pages. It will
+ * return an array of numbers and "...", where the numbers are the
+ * pages and the "..." are the dots in between.
+ *
+ * To handle a high amount of pages, it segments the pagination into
+ * three parts: the left segment, the middle segment and the right
+ * segment.
+ *
+ * here is an example of a 10 page pagination with 1 neighbour. the
+ * number wrapped in * is the current page
+ *
+ * *1* 2 ... 9 10
+ * 1 *2* 3 ... 9 10
+ * 1 2 *3* 4 ... 9 10
+ * 1 2 3 *4* 5 ... 9 10
+ * 1 2 3 4 *5* 6 ... 9 10
+ * 1 2 ... 5 *6* 7 ... 9 10
+ *
+ * @param pages the amount of pages
+ * @param currentPage the page we are currently on
+ * @param neighbours the amount of neighbours to the left and right of the current page, start and end defaults to 1
+ * @returns an array of numbers and "..."
+ */
+const describePagination = ({
+  pages,
+  currentPage: current,
+  neighbours = 1,
+}: {
+  pages: number;
+  currentPage: number;
+  neighbours?: number;
+}): PaginationShape => {
+  if (current < 1) return [];
+  if (pages < 1) return [];
+  if (current > pages) return [];
+  if (neighbours < 0) return [];
+
+  /**
+   * considering this pagination example
+   * 1 2 ... 5 *6* 7 ... 9 10
+   *
+   * the variables will be set as follows
+   *
+   * pages = 10
+   * current = 6
+   * neighbours = 1
+   *
+   * activeSegmentLeft = 5
+   * activeSegmentRight = 7
+   * activeSegment = [5, 6, 7]
+   * activeSegmentCount = 3
+   *
+   * startSegmentCount = 2
+   * startSegmentLeft = 1
+   * startSegmentRight = 2
+   * startSegment = [1, 2, "..."]
+   *
+   * endSegmentLeft = 9
+   * endSegmentRight = 10
+   * endSegmentCount = 2
+   * endSegment = ["...", 9, 10]
+   *
+   * and this this function will return
+   * [1, 2, "...", 5, 6, 7, "...", 9, 10]
+   */
+
+  // active segment
+  const currentLeft = current - neighbours;
```

Comment:
I would suggest `leftmostNeighbour` here (and `rightmostNeighbour` below, respectively)?

(don't forget to update this in the comments as well, if you change it)

Reply:
You mean `currentLeft` should be `leftmostNeighbour`, right? I'm just asking because there are no comments that I can update...

Reply:
Changed it as we discussed in person https://github.com/direktiv/direktiv-ui/pull/412/commits/02902336c694f2ebc66663d3fe4e1af8e27ea59e